core/src/streams.rs

```diff
@@ -331,6 +341,152 @@ impl<'gc> NetStream<'gc> {
         self.0.read().buffer.len()
     }
 
+    pub fn time(self) -> f64 {
+        self.0.read().stream_time
+    }
+
+    /// Queue a seek to be executed on the next frame tick.
+    ///
+    /// `offset` is in milliseconds.
+    pub fn seek(self, context: &mut UpdateContext<'_, 'gc>, offset: f64, notify: bool) {
+        self.0.write(context.gc_context).queued_seek_time = Some(offset);
+        StreamManager::activate(context, self);
+
+        if notify {
+            let trigger = AvmString::new_utf8(
+                context.gc_context,
+                format!("Start Seeking {}", offset as u64),
+            );
+            self.trigger_status_event(
+                context,
+                vec![
+                    ("description", trigger),
+                    ("level", "status".into()),
+                    ("code", "NetStream.SeekStart.Notify".into()),
+                ],
+            );
+        }
+    }
+
+    /// Seek to a new position in the stream.
+    ///
+    /// All existing audio will be paused. The stream offset will be snapped to
+    /// the prior keyframe. If the stream is playing then new tag processing
```

Comment:
> The stream offset will be snapped to the prior keyframe.

Is it always "prior", even when seeking forwards?
I'm not saying it's not, it's just not immediately obvious from the first reading.

Reply:
You're right, it goes to the next keyframe in the direction of the seek.